File: 백준/Bronze/2480. 주사위 세개/주사위 세개.go
package main

import (
    "fmt"
    "bufio"
    "os"
)

func main() {
    reader := bufio.NewReader(os.Stdin)
    resultCount := make([]int, 7)
    var times, maxNum, manyNum, maxCount, awards int

    for i := 0; i < 3; i++ {
        fmt.Fscanf(reader, "%d", &times)
        resultCount[times]++

        if maxNum < times {
            maxNum = times
        }
        
        if maxCount < resultCount[times] {
            maxCount = resultCount[times]
        }

        if maxCount == resultCount[times] {
            manyNum = times
        }
    }

    if maxCount == 1 {
        awards = maxNum * 100
    }

    if maxCount == 2 {
        awards = 1000 + manyNum * 100
    }

    if maxCount == 3 {
        awards = 10000 + maxNum * 1000
    }

    fmt.Println(awards)
    
}

File: 백준/Bronze/2480. 주사위 세개/주사위 세개.c
#include <stdio.h>

int main() {
    int resultCount[7] = {0};
    int times = 0;
    int maxNum = 0;
    int manyNum = 0;
    int maxCount = 0;
    int awards = 0;

    for (int i = 0; i < 3; i++) {
        scanf("%d", &times);
        resultCount[times] ++;

        if (maxNum < times) {
            maxNum = times;
        }

       if (maxCount < resultCount[times] ) {
           maxCount = resultCount[times];
       }

       if (maxCount == resultCount[times]) {
           manyNum = times;
       }
    }

    if (maxCount == 1) {
        awards = maxNum * 100;
    }

    if (maxCount == 2) {
        awards = 1000 + manyNum * 100;
    }

    if (maxCount == 3) {
        awards = 10000 + maxNum * 1000;
    }

    printf("%d", awards);
    
    return 0;
}